cucumber-core/src/main/java/io/cucumber/core/options/CucumberPropertiesParser.java

```diff
@@ -195,10 +203,29 @@ private static <T> Function<String, Collection<T>> splitAndMap(Function<String,
                 .collect(toList());
     }
 
-    private static Stream<Collection<FeatureWithLines>> parseRerunFile(String property) {
+    private static Set<String> listRerunFiles(String path) {
+        Set<String> fileList = new HashSet<>();
+        try {
+            Files.walkFileTree(Paths.get(path), new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+                    if (!Files.isDirectory(file)) {
+                        fileList.add(path + "/" + file.getFileName().toString());
```

Comment:
If I'm not mistaken, the `Path file` is a complete path the the rerun file. So you can directly add it to the the `fileList`.